src/receiving.rs

```diff
@@ -195,21 +195,20 @@ impl Receiver {
         &self,
         tweak_data: &PublicKey,
         pubkeys_to_check: Vec<XOnlyPublicKey>,
-    ) -> Result<HashMap<Label, HashSet<SecretKey>>> {
+    ) -> Result<HashMap<Label, Vec<Scalar>>> {
         let secp = secp256k1::Secp256k1::new();
-        let B_spend = &self.spend_privkey.public_key(&secp);
         let ecdh_shared_secret = self.calculate_shared_secret(tweak_data)?;
 
-        let mut my_outputs: HashMap<Label, HashSet<SecretKey>> = HashMap::new();
+        let mut my_outputs: HashMap<Label, Vec<Scalar>> = HashMap::new();
         let mut n: u32 = 0;
         while my_outputs.len() == n as usize {
-            let t_n: Scalar = calculate_t_n(&ecdh_shared_secret, n)?;
-            let P_n: PublicKey = calculate_P_n(&B_spend, t_n)?;
+            let t_n = calculate_t_n(&ecdh_shared_secret, n)?;
```

Comment:
Nit: I'd rather keep the explicit type annotation here, so that we don't have to check the calculate_t_n() signature when reading

Reply:
done